src/UCTNodePointer.cpp

```diff
@@ -26,17 +26,18 @@
 #include "UCTNode.h"
 
 UCTNodePointer::~UCTNodePointer() {
-    if (is_inflated()) {
-        delete read_ptr();
+    auto v = std::atomic_exchange(&m_data, INVALID);
+    if (is_inflated(v)) {
+        delete read_ptr(v);
     }
 }
 
 UCTNodePointer::UCTNodePointer(UCTNodePointer&& n) {
-    if (is_inflated()) {
-        delete read_ptr();
+    auto nv = std::atomic_exchange(&n.m_data, INVALID);
```

Comment:
What's the use case where setting n.m_data to invalid would be needed?

Reply:
This is the move constructor so we have to make sure the original instance no longer owns the pointer hence have to overwrite it with something.  Otherwise the destructor of the original instance will delete it.  Did I get your question correctly?

Reply:
OK!